### ISSUE TITLE
Use Chrome DevTools Protocol to Work Around Photos API issues.

### ISSUE THREAD

**Issue by gilesknap** [gilesknap/gphotos-sync#271]
Look into using google takeout to extract photos from the library to overcome:

- lack of GPS
- compressed video files
- compressed original quality images

Note that @mholt has already done work in this area https://github.com/mholt/timeliner/issues/38#issuecomment-738533821

This is more of an exit strategy than support for gphotos-sync but could be used in parallel in some clumbsy fashion (I currently use gphotos-sync to keep an eye on accidental deletions, rather than a full backup)

**Comment by gilesknap** [comment]
Now this seems better than supporting takeout.
https://github.com/perkeep/gphotos-cdp

If we use the existing gphotos-sync code to catalogue the photos library using Google API but replace its download code with the above then we might just get around many of the API nasty restrictions.

@mpl I just found your project and it looks like a good workaround to many of the crippling limitations of Google Photos API. Do you think that we could create some API such that I can drive the Chrome DevTools Protocol from Python?

**Comment by mpl** [comment]
@gilesknap I haven't looked, but I would be surprised if there isn't already a python library to drive the Chrome DevTools Protocol. Have you checked?

**Comment by gilesknap** [comment]
@mpl yes it appears there is https://pypi.org/project/chrome-devtools-protocol/ - Sorry, I made this comment without any research as a reminder to look at it :-)

**Comment by gilesknap** [comment]
Closing as won't fix. I need Google to step up and make the API work in a useful fashion. Not chase a bunch of hacks to get around the API limitations.
